models/src/amount.rs

```diff
@@ -11,6 +11,13 @@ pub const AMOUNT_DECIMAL_FACTOR: u64 = 1_000_000_000;
 #[derive(Clone, Copy, Debug, PartialEq, Eq, Ord, PartialOrd, Default)]
 pub struct Amount(u64);
 
+impl std::ops::Add for Amount {
```

Comment:
We choose to use saturating add or checked add and not just add

@damip I don't think we documented why and when to use one instead of the other ?

Reply:
We could have an overflow that make the node panic :thinking:
Alright I'll use checked_add in this case

Reply:
Sadly, this trait is needed by checked_add

Reply:
We always use saturating_XX and checked_XX on amounts because they are critical. Let's not use std::ops::Add

Reply:
done